### PR TITLE
GEOPY-857

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-857 - clip by extent fails with memory error on large file**
@sebhmg I added an argument to clear attributes on copy. Should solve your issue... unless you can't load the original object with its data. Otherwise will need to slice at the source.

### CODE REVIEW

File: tests/clip_vertex_data_test.py [hunk 1]
```diff
@@ -136,10 +136,15 @@ def test_clip_groups(tmp_path):
                 },
             }
         )
+
         with Workspace(tmp_path / r"testClipPoints_copy.geoh5") as new_workspace:
-            group_a.copy_from_extent(extent, parent=new_workspace)
+            group_a.copy_from_extent(extent, parent=new_workspace, clear_cache=True)
```
Comment (sebhmg) [inline tests/clip_vertex_data_test.py:141]: **suggestion (tests):** have a test test with `clear_cache=False`

and also tests for `Workspace.copy_to_parent()` and  implementation `Entity.copy()` with `clear_cache=True` and `clear_cache=False`

File: geoh5py/groups/group.py [hunk 2]
```diff
@@ -75,7 +75,11 @@ def add_comment(self, comment: str, author: str | None = None):
             self.comments.values = self.comments.values + [comment_dict]
 
     def copy_from_extent(
-        self, bounds: np.ndarray, parent=None, copy_children: bool = True
+        self,
+        bounds: np.ndarray,
+        parent=None,
+        copy_children: bool = True,
+        clear_cache: bool = False,
```
Comment (sebhmg) [inline geoh5py/groups/group.py:82]: **issue (doc):** obsolete doc of parameters

The 3 last parameters are not documented, while the doc mentions non existing parameter `attributes`

File: geoh5py/objects/object_base.py [hunk 3]
```diff
@@ -220,7 +220,11 @@ def default_type_uid(cls) -> uuid.UUID:
         ...
 
     def copy_from_extent(
-        self, bounds: np.ndarray, parent=None, copy_children: bool = True
+        self,
+        bounds: np.ndarray,
+        parent=None,
+        copy_children: bool = True,
+        clear_cache: bool = False,
```
Comment (sebhmg) [inline geoh5py/objects/object_base.py:227]: **issue (doc):** obsolete doc of parameters

The 3 last parameters are not documented, while the doc mentions non existing parameter `attributes`

File: geoh5py/objects/surveys/direct_current.py [hunk 4]
```diff
@@ -103,13 +103,20 @@ def ab_map(self) -> dict | None:
...
+        clear_cache: bool = False,
+        **kwargs,
+    ):
         """
         Function to copy a survey to a different parent entity.
 
         :param parent: Target parent to copy the entity under. Copied to current
             :obj:`~geoh5py.shared.entity.Entity.parent` if None.
         :param copy_children: Create copies of all children entities along with it.
+        :param clear_cache: Clear array attributes after copy.
```
Comment (sebhmg) [inline geoh5py/objects/surveys/direct_current.py:119]: **suggestion:** remove ambiguity in the doc: does it clear attributes on `self` or on the `copy`

Also should the doc says clears the attributes _cache_ array?

I do not quite understand where attributes are stored if cleared in memory and the file has not been saved to disk yet,

**Review by sebhmg — CHANGES_REQUESTED** [review]
thanks !
please see some suggestions and questions

**Comment by sebhmg** [comment]
I ma testing with on a 3 GB geoh5, and it eats 11 GB right away, before being killed with exit code 137

image

**Comment by domfournier** [comment]
> I am testing with on a 3 GB geoh5, and it eats 11 GB right away, before being killed with exit code 137
> 
> image

So I guess you can't even load the object with it's data in memory. Will have to slice one data at a time.

**Pull request closed without merging**